core/eth/contracts/FluxAggregator.go

```diff
@@ -0,0 +1,181 @@
+	PaymentAmount  *big.Int
+	MinAnswerCount *big.Int
+	MaxAnswerCount *big.Int
+	RestartDelay   *big.Int
+	Timeout        *big.Int
+	Address        common.Address
+}
+
+type LogAnswerUpdated struct {
+	Current   *big.Int
+	RoundID   *big.Int
+	Timestamp *big.Int
+	Address   common.Address
+}
+
+func NewFluxAggregator(ethClient eth.Client, address common.Address) (FluxAggregator, error) {
+	contract, err := eth.GetV6Contract(eth.FluxAggregatorName)
+	if err != nil {
+		return nil, err
+	}
+	connectedContract := eth.NewConnectedContract(contract, ethClient, address)
+	return &fluxAggregator{connectedContract, ethClient, address}, nil
+}
+
+func (fa *fluxAggregator) SubscribeToLogs(fromBlock *big.Int) (LogSubscription, error) {
+	var (
+		filterQuery = ethereum.FilterQuery{
+			FromBlock: fromBlock,
+			Addresses: utils.WithoutZeroAddresses([]common.Address{fa.address}),
+			Topics:    [][]common.Hash{{models.AggregatorNewRoundLogTopic20191220, models.AggregatorRoundDetailsUpdatedLogTopic20191220, models.AggregatorAnswerUpdatedLogTopic20191220}},
+		}
+		chLogs    = make(chan MaybeDecodedLog)
+		chRawLogs = make(chan eth.Log)
+	)
+
+	subscription, err := fa.ethClient.SubscribeToLogs(chRawLogs, filterQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		defer close(chLogs)
+		for {
+			select {
+			case err := <-subscription.Err():
+				chLogs <- MaybeDecodedLog{nil, err}
+
+			case rawLog, stillOpen := <-chRawLogs:
+				if !stillOpen {
+					select {
+					case err := <-subscription.Err():
+						chLogs <- MaybeDecodedLog{nil, err}
+					default:
+					}
+					return
+				}
+
+				switch rawLog.Topics[0] {
+				case models.AggregatorNewRoundLogTopic20191220:
+					var decodedLog LogNewRound
+					err = fa.UnpackLog(&decodedLog, "NewRound", rawLog)
+					decodedLog.Address = fa.address
+					chLogs <- MaybeDecodedLog{decodedLog, err}
+
+				case models.AggregatorRoundDetailsUpdatedLogTopic20191220:
+					var decodedLog LogRoundDetailsUpdated
+					err = fa.UnpackLog(&decodedLog, "RoundDetailsUpdated", rawLog)
+					decodedLog.Address = fa.address
+					chLogs <- MaybeDecodedLog{decodedLog, err}
+
+				case models.AggregatorAnswerUpdatedLogTopic20191220:
+					var decodedLog LogAnswerUpdated
+					err = fa.UnpackLog(&decodedLog, "AnswerUpdated", rawLog)
+					decodedLog.Address = fa.address
+					chLogs <- MaybeDecodedLog{decodedLog, err}
+				}
+			}
+		}
+	}()
+
+	return &logSubscription{subscription, chLogs}, nil
+}
+
+var dec10 = decimal.NewFromInt(10)
+
+// Price returns the current price at the given aggregator address.
+func (fa *fluxAggregator) LatestAnswer(precision int32) (decimal.Decimal, error) {
+	var result *big.Int
+	err := fa.Call(&result, "latestAnswer")
+	if err != nil {
+		return decimal.Decimal{}, errors.Wrap(err, "unable to encode message call")
+	}
+	raw := decimal.NewFromBigInt(result, 0)
+	precisionDivisor := dec10.Pow(decimal.NewFromInt32(precision))
+	return raw.Div(precisionDivisor), nil
```

Comment:
Why not use the second argument to `NewFromBigInt`?

Reply:
I'm going to leave this the way it is for now, since this is how the code was already dealing with it, but we can look into this in a subsequent story.